vcpkg/users/binarycaching-troubleshooting.md

```diff
@@ -0,0 +1,164 @@
+[DEBUG]   vcpkg_from_github|b743742296a114ea1b18ae99672e02f142c4eb2bef7f57d36c038bedbfb0502f
+[DEBUG]   vcpkg_replace_string|d43c8699ce27e25d47367c970d1c546f6bc36b6df8fb0be0c3986eb5830bd4f1
+[DEBUG] </abientries>
+```
+
+You might notice the `triplet_abi` entry has three hashes: the hash of the file content of the `x86-windows` triplet, the `windows.cmake` toolchain, and the compiler hash. Theses hashes would change if you decide to target a different platform.
+
+### The hashes of the files used to build your library (`portfile.cmake`, `*.patch`, or `vcpkg.json`) change between runs
+
+This usually means the baseline of your libraries is not fixed. Frequently, the problem occurs when you clone the latest repository from vcpkg, which will pull in the updated baseline of libraries.
+
+Possible solutions:
+
+* (recommended) The copy of the vcpkg catalog must be pinned to a specific commit. When you run `git clone vcpkg`, make sure you run `git checkout [commit_id]` or add vcpkg as a submodule in your project. The reason is the ABI of a library is calculated based on the hashes of scripts and ports in the vcpkg repository. Pulling the latest changes will likely cause frequent rebuilds.
+* Set `builtin-baseline` in your [vcpkg manifest](../reference/vcpkg-json.md#builtin-baseline).
+
+### ABI hash of vcpkg CMake helper functions changes between runs
+
+Internal vcpkg helper functions like vcpkg_from_github, vcpkg_copy_pdbs, vcpkg_fixup_pkgconfig, etc. are used in the ABI calculation. If you are getting new ABI hashes for these ports, this is probably also caused by pulling the latest vcpkg repo. This issue will be addressed in a future update when these functions will be separately versioned.
+
+Possible solutions:
+
+* (recommended) The copy of the vcpkg catalog must be pinned to a specific commit. When you run `git clone vcpkg`, make sure you run `git checkout [commit_id]` or add vcpkg as a submodule in your project. The reason is the ABI of a library is calculated based on the hashes of scripts and ports in the vcpkg repository. Pulling the latest changes will likely cause frequent rebuilds.
+
+### Version of CMake, PowerShell, or your compiler changes between runs
+
+In these cases, vcpkg is behaving correctly. It is expected for toolchain changes to cause ABI hash changes as well. Rebuilds can be expensive, so we suggest the following:
+
+* Continuous integration systems: Some systems such as GitHub Hosted Runners update the underlying image periodically. Updates may change your C/C++ compiler or CMake version, which will cause a rebuild. If possible, choose a versioned stable image.
+* MSVC compiler updates: Major, minor and patch updates to the MSVC compiler will cause your library ABI hash to change. To minimize this, turn off automatic Visual Studio updates.
+
+If the above options do not work, consider the following workarounds:
+
+* Use `vcpkg export` to package your libraries instead
```

Comment:
```suggestion
* Use [`vcpkg export`](../commands/export.md) to produce a standalone archive of your dependencies instead of restoring them from a manifest.
```